### ISSUE TITLE
Feature Request: Setting Goals using a Question/Query

### ISSUE THREAD

**Issue by Knight0fNever** [metabase/metabase#6514]
Hi,
Thanks for the awesome tool! It's helped our small business greatly.

Would it be possible to be able to use a question or a native query to automatically set the goal in a progress bar?

Thanks!

**Comment by benjR** [comment]
Same in graphs, the goal is just a static line today and with date related graphs, you never have a static goal, you should be able to change it from a month to another for example

**Comment by vincentsjtu** [comment]
definitely great idea, i think this dynamic goal is very useful

**Comment by neilmitchell-goodson** [comment]
+1 here; a static goal is less useful than a dynamic one.

**Comment by flamber** [comment]
Duplicate of #4412
PR #5077

**Comment by noblemavely** [comment]
+1 for this feature request

**Comment by brasrox** [comment]
+1 please i need it!

**Comment by ChristineChetty11** [comment]
+1!!!!!!

**Comment by andrefelipefraga** [comment]
Please, I also need this feature

**Comment by bob6664569** [comment]
+1

**Comment by MarcioVaka017** [comment]
Without it the goal feature is simply useless...
Every other problem I have found in Metabase has a workaround, but I really couldn't find one for this.
Will be using a line chart (goal) over a bar chart (actual results), but it really isn't what we need.

**Comment by itai-codefresh** [comment]
Up voting this as well.
Its critical for us esoecially in line charts.

I see its scheduled to version 0.38?

**Comment by Man-in-Black** [comment]
I'm also interested in this feature and hope for an implementation soon.

**Comment by ccalby** [comment]
Definitely a beneficial feature. Waiting on this too!

**Comment by EasyPat82** [comment]
This feature was published to be a small win on 0.38 however since then many people are still requesting this and keep being relayed to this #6514. But the feature is still not available, right? I am unable to do this still.

**Comment by flamber** [comment]
This issue is still open, meaning it is not fixed/implemented. When an issue is closed, then look in the sidebar for a milestone.
There were several things that was added to roadmap, which didn't make it into a release yet.

**Comment by Erin007** [comment]
This feature is exactly what I need +1 for implementing soon, please 🙏

**Comment by ashishorkalra** [comment]
Is there a possible/rough timeline around this feature request? I feel this is requested by quite a number of people here, including me and my company.

**Comment by ashishorkalra** [comment]
@flamber it seems, it is the most request feature in the category of `Visualisation/Chart Setting`. I am not sure how many categories are there, but it seems, it is very well requested by the community. Honestly, we also need that badly.

**Comment by faq885** [comment]
It will be amazing to have this :)

**Comment by MaximeMathey** [comment]
+1 on this feature :)

**Comment by luizarakaki** [comment]
We are currently planning this feature and would appreaciate your feedback.

In you case, where does the goal come from?

Do you have it in the same question (as another field)? Is it in another table in the same database? Or is it elsewhere in another database?

**Comment by EasyPat82** [comment]
Would be great to finally see this in production. 
In my case, the goal is either in the same db in a different table, or in the same table. But I can imagine some users would want the outcome of a question to be the goal?

**Comment by superbhopkins** [comment]
Hi @luizarakaki 

From our side, our goals are part of a calculation based on the same data source. For example, we compare the current bookings a customer has this week vs the average bookings that an arbitrary customer would have in a week (this is calculated as a rolling average).

**Comment by fabioknoedt** [comment]
@luizarakaki we would set the goal based on a different field on the same question/query. Having the change to bring from another table or another database will for sure make it easier for more teams though.

**Comment by luizarakaki** [comment]
Thanks for the feedback. We are moving forward with the designs, let me know what you think:

![image](https://github.com/metabase/metabase/assets/32561114/14f83c33-7dd0-4bf7-bd00-be0bcedc502b)
- You can add multiple goals.
- For each goal, you can choose static (type your goal) or dynamic (pick a column from the same question)

When hovering the line/bar you get this tooltip
![image](https://github.com/metabase/metabase/assets/32561114/ceadbe08-c214-4565-9603-004fbc1aa9b5)

The same principle applies to Trend charts
![image](https://github.com/metabase/metabase/assets/32561114/a53b2fe0-f5a4-4f59-a4ad-91221259695e)

---

- Does this solve the problem? Or getting the value from another question is essential?
- Anything we should change in the designs?

If you are up to a chat about this, let's talk. Book a time [here](https://calendly.com/luiz-arakaki/15-min)

**Comment by superbhopkins** [comment]
For our uses this would work. Is this planned also for the Gauge chart? How would the target look on the gauge?

**Comment by fabioknoedt** [comment]
@luizarakaki 

> Does this solve the problem? Or getting the value from another question is essential?
Yes, that would solve the problem for us.

> Anything we should change in the designs?
It looks fine to me. Good job!

**Comment by luizarakaki** [comment]
@superbhopkins Gauge designs still WIP. But it would look very similar: you define ranges based on other fields.

**Comment by superbhopkins** [comment]
@luizarakaki got it. That is a perfect solution for us. Looking forward to it.

**Comment by paoliniluis** [comment]
@dguzman1012 please check our public roadmap

**Comment by jlitke** [comment]
Hello! Curious on a planned release date for this? I do not see it on the public roadmap (this is where we were told to look). 

Thanks!

**Comment by Terterola** [comment]
> Gauge designs still WIP. But it would look very similar: you define ranges based on other fields.

@luizarakaki,

I've just started experimenting Metabase, but fixed values to Gauge and Progress Bar really makes its use too limited and, in fact, I guess, it doesn't even make sense in most business context. Goals may vary a lot, from month to month, from a user to another user, depending on the filter applied, and so on. Hope the flexibility to set dynamic goals can be available soon. 

In my case, I have tables prepared for BI presentation, so I can have in the same row a column with the VALUE and another column with what would be the GOAL. But things could get even more flexible if we could write any SQL to collect and organize the necessary data. 

BTW, vi que você é brasileiro, de São Paulo, isso? Sou de Porto Alegre. Se achar válido, podemos trocar figurinhas mais facilmente para validar essa funcionalidade. Seria algo muito importante para eu conseguir sugerir o uso do Metabaes para os meus clientes. Se possível, aguardo notícias sobre o andamento deste desenvolvimento. Abraço!

**Comment by lucascbarbosa** [comment]
Same here. Se quiserem conversar sobre só chamar!

**Comment by paoliniluis** [comment]
Just for reference: the goal configuration is a front end concept today, making it dynamic means that we need to take it entirely to the backend, so it’s not a simple change

**Comment by luizarakaki** [comment]
We are now working to improve our viz infra that will make this project easier to implement. As this is a larger refactor, it is taking longer than anticipated. This feature is not scheduled for v50.

cc @cdeweyx

**Comment by lcoudrot** [comment]
Hope release as soon as possible, thanks :)

**Comment by J2M5Raiden** [comment]
Any update on progress of this development or a anticipated delivery date?

**Comment by paoliniluis** [comment]
@J2M5Raiden please check our roadmap https://www.metabase.com/roadmap

**Comment by hss-iullah** [comment]
Metabase has zero support for dynamic expressions in the tool. Not one of the visualisations support it. BI tools have supported this since the 2000's and is the most basic need for any BI tool to succeed today. If they get dynamic expressions into the product I'm sure they'll blow away the competition!

**Comment by tdowns** [comment]
+1 for this, I'd like to set goals based on a query!

**Comment by migs35323** [comment]
+1

**Comment by panchocean** [comment]
+1

**Comment by guilherme-poli** [comment]
+1

**Comment by zberkom** [comment]
+1

**Comment by zberkom** [comment]
Any update on this ticket?

**Comment by Axxoul** [comment]
I will pay 100 USD for this to be implemented

**Comment by dragonsahead** [comment]
> ⬇️ Please click the 👍 reaction instead of leaving a +1 or update?

we evaluate the need of issues based on the amount of thumbs up the issue has

**Comment by claywalker-lumos** [comment]
@paoliniluis Thanks! which issue has greater than 622 thumbs up? Are they stack ranked anywhere with public vis?

**Comment by hss-iullah** [comment]
The problem is that they are releasing less useful “features” that haven’t been upvoted for. So dynamic expressions will never make it. Just like new pivot tables which have been promised for almost 2 years!

**Comment by Denis2803** [comment]
+1

**Comment by lucashamaguchi** [comment]
+1

**Comment by dragonsahead** [comment]
Edited the first comment since progress bar is coming on 57

**Comment by panchocean** [comment]
Thank you! But I think that setting dynamic goals in a gauge is more critical, since you can always set a dynamic goal in the denominator and interpret the percentage of accomplishment in progress bars.

**Comment by Paulbaobab** [comment]
+1

**Comment by CassiusNem** [comment]
+1
Really Metabase? This feature is awesome why don't you care? Pleaseeeeeee

**Comment by alexyarosh** [comment]
As Luis mentioned, in the Progress bar viz, you can [accomplish this](https://www.metabase.com/docs/latest/questions/visualizations/progress-bar#use-another-querys-result-as-the-goal) now with v57 🎉 !

**Comment by claywalker-lumos** [comment]
This is a great candidate for some internal feedback. We were considering a metabase contract (ie migrating off the free version) but opted instead for Sigma computing. Thank you for taking action on this and I wish you the best.

**Comment by captvegith** [comment]
+1

**Comment by artur-w** [comment]
+1
